core/api/src/app/authentication/totp.ts

```diff
@@ -23,17 +31,70 @@ export const validateTotpRegistration = async ({
   totpRegistrationId,
   userId,
 }: {
-  authToken: AuthToken
+  authToken: AuthToken | null
   totpCode: TotpCode
   totpRegistrationId: TotpRegistrationId
   userId: UserId
 }): Promise<User | ApplicationError> => {
+  if (authToken) {
+    return validateTotpRegistrationWithToken({
+      authToken,
+      totpCode,
+      totpRegistrationId,
+      userId,
+    })
+  }
+
+  return validateTotpRegistrationWithoutToken({
+    totpCode,
+    totpRegistrationId,
+    userId,
+  })
+}
+
+export const validateTotpRegistrationWithoutToken = async ({
```

Comment:
```suggestion
const validateTotpRegistrationWithoutToken = async ({
```

this doesn't need to be exported